CMakeLists.txt

```diff
@@ -31,141 +32,127 @@ message(STATUS "**** Build type = ${CMAKE_BUILD_TYPE}")
 # set(CMAKE_BUILD_TYPE Debug)
 # set(CMAKE_VERBOSE_MAKEFILE 1)
 
-
 ################################
 # General compiler configuration
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_EXTENSIONS OFF) # -std=c++11 instead of -std=gnu++11
-
-add_definitions(-DHPCOMBI_HAVE_CONFIG)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_EXTENSIONS OFF) # -std=c++14 instead of -std=gnu++14
 
 message(STATUS "*** Compiler id is ${CMAKE_CXX_COMPILER_ID}")
-if ( CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang" )
-  # Workaround of CMAKE bug https://stackoverflow.com/questions/47213356/
-  set(CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS} -std=c++11)
-  add_compile_options(-std=c++11 -Wall -g -pg)
-endif()
-
+# See https://stackoverflow.com/questions/47213356/ for discussion
+cmake_policy(SET CMP0067 NEW)
 
 ##################################
 # Checks for compiler capabilities
 include(CheckCXXSymbolExists)
 include(CheckIncludeFileCXX)
 include(CheckCXXSourceCompiles)
 include(CheckCXXCompilerFlag)
+include(CheckCXXSourceRuns)
 
-## Check for static lcm
-check_include_file_cxx("experimental/numeric" HPCOMBI_HAVE_EXPERIMENTAL_NUMERIC)
-if (HPCOMBI_HAVE_EXPERIMENTAL_NUMERIC)
-  check_cxx_source_compiles(
-        "
-        #include <experimental/numeric>
-        static_assert(std::experimental::lcm(4, 6) == 12, \"Buggy lcm\");
-        int main() { }
-        "
-        HPCOMBI_HAVE_EXPERIMENTAL_NUMERIC_LCM)
-endif (HPCOMBI_HAVE_EXPERIMENTAL_NUMERIC)
-
-## Check for buggy constexpr support G++ 5.0
-check_cxx_source_compiles(
-  "
-  using T = int;
-  constexpr int exec(T f()) { return f(); }
-  constexpr int foo() { return 1; }
-  static_assert(exec(foo) == 1, \"Failed exec\");
-  int main() {}
-  "
-  HPCOMBI_CONSTEXPR_FUN_ARGS)
+check_cxx_compiler_flag('-funroll-loops' HPCOMBI_HAVE_REQUIRED_COMPILER_FLAGS)
+check_cxx_compiler_flag('-flax-vector-conversions' HPCOMBI_HAVE_REQUIRED_COMPILER_FLAGS)
+
+if (NOT HPCOMBI_HAVE_REQUIRED_COMPILER_FLAGS)
+  message(FATAL_ERROR "HPCombi requires a C++ compiler accepting the flags: -funroll-loops + -flax-vector-conversions")
+endif()
+
+add_compile_options(-funroll-loops -flax-vector-conversions)
 
-check_cxx_compiler_flag('-mavx' HPCOMBI_HAVE_FLAG_AVX)
-#check_cxx_compiler_flag('-march=native' HPCOMBI_HAVE_FLAG_NATIVE)
-#check_cxx_compiler_flag('-mavx2' HPCOMBI_HAVE_FLAG_AVX2)
-#check_cxx_compiler_flag('-mavx512bw' HPCOMBI_HAVE_FLAG_AVX512BW)
```

Comment:
I recommend also checking for `-fopenmp-simd` and if it is found, adding it and `-DSIMDE_ENABLE_OPENMP` ([generates better autovectorization](https://github.com/simd-everywhere/simde/#openmp-4-simd))